app/services/samples/metadata/update_service.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+module Samples
+  module Metadata
+    # Service used to Update Samples::Metadata
+    class UpdateService < BaseService
+      SampleMetadataUpdateError = Class.new(StandardError)
+      attr_accessor :sample, :metadata, :analysis_id
+
+      def initialize(project, sample, user = nil, params = {})
+        super(user, params)
+        @project = project
+        @sample = sample
+        @metadata = params['metadata']
+        @analysis_id = params['analysis_id']
+      end
+
+      def execute # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+        authorize! sample.project, to: :update_sample?
+        metadata_fields_update_status = { updated: [], not_updated: [] }
+
+        validate_sample_in_project
+
+        validate_metadata_param
+
+        transform_metadata_keys
+
+        @metadata.each do |key, value|
```

Comment:
We should probably move this block into it's own method as well which will return the ` metadata_fields_update_status`

Reply:
Updated in f28954cde8515abffe12e9ac9b777ca728ecce83